tinygrad/engine/realize.py

```diff
@@ -37,7 +37,9 @@ def get_linearizer(renderer:Renderer, ast:Tuple[LazyOp, ...]) -> Linearizer:
         k = timed[0][1]
         if logkerns is not None and logkerns_level > 1: logkerns.writelines([f"{(lin.ast, lin.applied_opts)}\n" for (_,lin,_) in timed[1:]])
   # TODO: check the correctness inline once compare_linearizer is in core
-  if logkerns is not None: logkerns.writelines([f"{(k.ast, k.applied_opts)}\n"])
+  if logkerns is not None:
+    logkerns.writelines([f"{(k.ast, k.applied_opts)}\n"])
+    logkerns.close()
```

Comment:
this should stay opened after writelines, there can be multiple calls to this function